wrapper/src/combo.c

```diff
@@ -72,8 +73,9 @@ void* combo_threadfunc(void* vmaf_thread_data)
     int h = thread_data->h;
     char* errmsg = thread_data->errmsg;
     void* user_data = thread_data->user_data;
-    const char* fmt = thread_data->fmt;
-    int n_subsample = thread_data->n_subsample;
+    enum VmafPixelFormat fmt = thread_data->fmt;
+    unsigned int n_subsample = thread_data->n_subsample;
+    bool use_color = thread_data->use_color;
```

Comment:
Would `use_color` also be added to all.c, and other functions like ssim.c, psnr.c? I am debating if adding it here now is pre-mature optimization. Clearly, with this you can do one pass and compute for all Y, U and V, and this is more efficient. But consider the alternative: do 3 passes of the video file, each one use a different read_frame callback function to pass Y, U and V. This is less efficient for now, but will result in less boiler-plate code. If we do this, we can leave speed optimization to the future.

Reply:
Why not just ask for each feature explicitly? `psnr_y`, `psnr_cb`, `psnr_cr` ...

Using a global `use_color` bool is a little confusing. Which features have color? Do I really have to extract all of them if I set this?

Reply:
2.

The global nature of use_color is related to the fact that we have one read_frame. We can rename this to "read_color". In fact, "read_color" is not tied to a specific feature, it just means that you have read the color planes and have them available for extracting color features. I am now adding further logic to decouple the two: extract psnr color when the corresponding feature field is also turned on, i.e., read_color && DO_PSNR_U.

Reply:
Update: let's actually get rid of use_color, read_vmaf_picture already can have that data there. It is up to the caller if a feature with chroma has to be calculated. For PSNR, it is a field: VMAF_FEATURE_MODE_SETTING_DO_CHROMA_PSNR. Similar fields can be added in the future if we want chroma in other features. This will also simplify read_vmaf_picture.

Reply:
Fixed this.